src/main/java/org/springframework/cloud/deployer/spi/cloudfoundry/CloudFoundryTaskLauncher.java

```diff
@@ -0,0 +1,620 @@
+        Map<String, String> environmentVariables = new HashMap<>(1);
+
+        try {
+            environmentVariables.put("SPRING_APPLICATION_JSON", new ObjectMapper().writeValueAsString(appDeploymentRequest.getDefinition().getProperties()));
+        }
+        catch (JsonProcessingException e) {
+            throw new RuntimeException(e);
+        }
+
+        return client.packages()
+            .stage(StagePackageRequest.builder()
+                .packageId(packageId)
+                .stagingDiskInMb(diskQuota(appDeploymentRequest))
+                .stagingMemoryInMb(memory(appDeploymentRequest))
+                .environmentVariables(environmentVariables)
+                .build())
+            .log("stream.stageDroplet")
+            .map(Droplet::getId)
+            .log("stream.dropletId");
+    }
+
+    private int diskQuota(AppDeploymentRequest request) {
+        return parseInt(
+            request.getDeploymentProperties().getOrDefault(DISK_PROPERTY_KEY, valueOf(properties.getDisk())));
+    }
+
+    private int memory(AppDeploymentRequest request) {
+        return parseInt(
+            request.getDeploymentProperties().getOrDefault(MEMORY_PROPERTY_KEY, valueOf(properties.getMemory())));
+    }
+
+    private TaskStatus mapTaskToStatus(GetTaskResponse getTaskResponse) {
+
+        switch (getTaskResponse.getState()) {
+            case SUCCEEDED_STATE:
+                return new TaskStatus(getTaskResponse.getId(), LaunchState.complete, null);
+            case RUNNING_STATE:
+                return new TaskStatus(getTaskResponse.getId(), LaunchState.running, null);
+            case PENDING_STATE:
+                return new TaskStatus(getTaskResponse.getId(), LaunchState.launching, null);
+            case CANCELING_STATE:
+                return new TaskStatus(getTaskResponse.getId(), LaunchState.cancelled, null);
+            case FAILED_STATE:
+                return new TaskStatus(getTaskResponse.getId(), LaunchState.failed, null);
+            default:
+                throw new IllegalStateException(
+                    "Unsupported CF task state " + getTaskResponse.getState());
+        }
+    }
+
+    /**
+     * Look up the applicationId for a given app and confine results to 0 or 1 instance
+     *
+     * @param client a {@link CloudFoundryClient} to work with
+     * @param name Name of the application to get the id for
+     * @return {@link Mono} with the application's id
+     */
+    private static Mono<String> getApplicationId(CloudFoundryClient client, String name) {
+
+        return requestListApplications(client, name)
+            .singleOrEmpty()
+            .map(Application::getId);
+    }
+
+    private static Mono<String> getReadyApplicationId(CloudFoundryClient client, String applicationId) {
+        return requestApplicationDroplets(client, applicationId)
+            .filter(resource -> org.cloudfoundry.client.v3.droplets.State.STAGED.equals(resource.getState()))
+            .next()
```

Comment:
Ensure the semantics are what you think they are here.  Do you really mean the _first_ `STAGED` droplet?  Or do you always expect to be a `.single()` `STAGED` droplet?